src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -508,6 +545,29 @@ private void updateWorkerStatusFromRequests()
             return;
         }
 
+        if ( getOwnBuilding().hasWorkerOpenRequests( worker.getCitizenData() ) )
+        {
+            for ( final IRequest request : getOwnBuilding().getOpenRequests( worker.getCitizenData() ) )
+            {
+                final IPlayerRequestResolver playerResolver = worker.getCitizenColonyHandler().getColony().getRequestManager().getPlayerResolver();
+                final IRetryingRequestResolver retryingRequestResolver = worker.getCitizenColonyHandler().getColony().getRequestManager().getRetryingRequestResolver();
+
+                if (playerResolver.holdsRequest(request.getId()) || retryingRequestResolver.holdsRequest(request.getId()))
```

Comment:
Again this can be done in a generic way without writing special case code for those two resolvers.
Go the manager route.

Reply:
yes, done